contracts/chain-adapters/Blast_Adapter.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./interfaces/AdapterInterface.sol";
+import "../external/interfaces/WETH9Interface.sol";
+
+// @dev Use local modified CrossDomainEnabled contract instead of one exported by eth-optimism because we need
+// this contract's state variables to be `immutable` because of the delegateCall call.
+import "./CrossDomainEnabled.sol";
+import "@eth-optimism/contracts/L1/messaging/IL1StandardBridge.sol";
+import "@eth-optimism/contracts/L1/messaging/IL1ERC20Bridge.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../libraries/CircleCCTPAdapter.sol";
+import "../external/interfaces/CCTPInterfaces.sol";
+
+/**
+ * @notice Contract containing logic to send messages from L1 to Blast. This is a modified version of the Optimism adapter
+ * that excludes the custom bridging logic. It differs from the Base Adapter in that it uses a special
+ * Blast contract to bridge WETH and DAI, which are yielding rebasing tokens on L2, WETH and USDB.
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Blast_Adapter is CrossDomainEnabled, AdapterInterface, CircleCCTPAdapter {
+    using SafeERC20 for IERC20;
+    uint32 public constant L2_GAS_LIMIT = 200_000;
+
+    WETH9Interface public immutable L1_WETH;
+
+    IL1StandardBridge public immutable L1_STANDARD_BRIDGE;
+
+    // Bridge used to get yielding version of ERC20's on L2.
+    address private L1_BLAST_BRIDGE = 0x3a05E5d33d7Ab3864D53aaEc93c8301C1Fa49115;
```

Comment:
nit: should we make this and the DAI address below immutable and passed in the constructor so this contract will work on testnet?